client/core/core.go

```diff
@@ -3910,19 +3930,24 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 		return nil, 0, codedError(walletErr, fmt.Errorf("FundOrder error for %s, funding quantity %d (%d lots): %w",
 			wallets.fromAsset.Symbol, fundQty, lots, err))
 	}
+
 	coinIDs := make([]order.CoinID, 0, len(coins))
 	for i := range coins {
 		coinIDs = append(coinIDs, []byte(coins[i].ID()))
```

Comment:
Here's where we need to insert the address. The `eth.coin` type will need to return the address here, but somehow the amounts for ID used for the database record. I recommend differentiating the two by adding a method to the `asset.Coin` interface, `RecoveryID() []byte`, which will be used for the database record and ultimately `FundingCoin`. For UTXO-based assets, the `RecoveryID()` will be identical to the `ID()`, but for eth, it will be different, so no upgrades are needed.

Reply:
If I follow correctly, insert the address at index 0 and leave the rest of the coins as is?

Reply:
eth's `FundOrder` would only ever return a single coin right?

Reply:
@JoeGruffins what do you think of https://github.com/decred/dcrdex/compare/master...buck54321:coin-recovery-id?expand=1 ?

Reply:
That looks like a reasonable solution.

Although I will say that I am still open to keeping the encodings consistent and client code simpler by just sending the "dead weight" funding coin serialization that has the amount in it, changing the `FromAddress` method of `Trade` to `FromCoin`, and having the server's asset backend's `AssetBalancer` decode it internally.

Reply:
Eh, that suggestion affects `QuoteAccount` and `BaseAccount` and all the server and book tracking.  Too late to rework that.  NVM

Reply:
Current commits looking alright with RecoveryID.  You can pop out the redemption funding commit, but we'll also get the discoveraccount/testing seed PR in first.